app/main.py

```diff
@@ -1,13 +1,25 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(file) for file in errors],
+        "path": file_path,
+        "status": (
+            "failed"
+            if any(file["filename"] == file_path for file in errors)
+            else "passed"
```

Comment:
The condition for setting the status to 'failed' seems incorrect. The current condition checks if any error's filename matches the given file_path, but the errors list contains dictionaries with keys 'line_number', 'column_number', 'text', and 'code', not 'filename'. You should verify the logic to ensure it correctly determines the status based on the presence of errors.